http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -167,17 +171,15 @@ final class LeoAppServiceInterp[F[_]: Parallel](
                     )
                 }
               } yield lastUsed.some
-            case (Some(FormattedBy.Galaxy), None) =>
+            case (Some(FormattedBy.Galaxy), Some(CromwellRestore(_))) =>
+              getAlreadyFormattedError(FormattedBy.Galaxy, "Cromwell", ctx.traceId)
```

Comment:
Can you use a constant for "Cromwell" (and "Galaxy" on line 178)?

So this code is saying that if you are trying to use a disk for a type that is different from the restore type, throw an error? It would be so nice if we could find a way to express this without explicitly listing every combination.

I'm also confused about the cases on line 176 that are mapping to Galaxy having formatted the disk. Shouldn't the second argument in both of those cases be GalaxyRestore?

Reply:
The case conditions on line 176 are actually same as before (see the red area lines from 174 to 180) with a new Cromwell condition to it. I am thinking that disks previously formatted by Custom or GCE apps can't be reused by any other app and hence the second argument doesn't matter. @Qi77Qi can you clarify the reasoning behind this case statement?

Reply:
Nit, but I honestly would prefer there's no `getAlreadyFormattedError`, but instead just a custom exception

`final case class AlreadyFomattedError(alreadyFormattedBy: FormattedBy, targetAppName: String, traceId: TraceId)` 

And have `F.raiseError(???)` in where `getAlreadyFormattedError` is currently

Reply:
Yeah that can be done. It would be cleaner as well